components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/core/BrandingPreferenceManagementService.java

```diff
@@ -83,37 +70,35 @@
     public BrandingPreferenceModel addBrandingPreference(BrandingPreferenceModel brandingPreferenceModel) {
 
         String tenantDomain = getTenantDomainFromContext();
-        /**
-         * Currently this API provides the support to only configure tenant wise branding preference for 'en-US' locale.
-         * So always use resource name as default resource name.
-         */
-        String resourceName = getDefaultResourceName();
-        // Check whether a branding resource already exists with the same name in the particular tenant to be added.
-        if (isResourceExists(BRANDING_RESOURCE_TYPE, resourceName)) {
-            throw handleException(Response.Status.CONFLICT, ERROR_CODE_CONFLICT_BRANDING_PREFERENCE, tenantDomain);
-        }
         String preferencesJSON = generatePreferencesJSONFromRequest(brandingPreferenceModel.getPreference());
         if (!BrandingPreferenceUtils.isValidJSONString(preferencesJSON)) {
             throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_BRANDING_PREFERENCE, null);
         }
 
+        BrandingPreference requestDTO, responseDTO;
         try {
-            InputStream inputStream = BrandingPreferenceUtils.generatePreferenceInputStream(preferencesJSON);
-            Resource brandingPreferenceResource =
-                    buildResourceFromBrandingPreference(brandingPreferenceModel, inputStream);
-            BrandingPreferenceServiceHolder.getBrandingPreferenceConfigManager()
-                    .addResource(BRANDING_RESOURCE_TYPE, brandingPreferenceResource);
-        } catch (ConfigurationManagementException e) {
-            throw handleConfigurationMgtException(e, ERROR_CODE_ERROR_ADDING_BRANDING_PREFERENCE, tenantDomain);
-        } catch (JsonProcessingException e) {
-            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_JSON_PROCESSING_EXCEPTION,
-                    e.getMessage());
-        } catch (UnsupportedEncodingException e) {
-            throw handleException(Response.Status.INTERNAL_SERVER_ERROR, ERROR_CODE_UNSUPPORTED_ENCODING_EXCEPTION,
-                    e.getMessage());
+            requestDTO = buildRequestDTOFromBrandingRequest(brandingPreferenceModel);
```

Comment:
Since this variable is only needed in this scope, move the definition here.

Reply:
Addreesed with https://github.com/wso2/identity-api-server/pull/345/commits/80cfa31001a7c2efa8b32467596dd2e92ea61750